controllers/propagator/replicatedpolicy_controller.go

```diff
@@ -346,12 +360,12 @@ func (r *ReplicatedPolicyReconciler) singleClusterDecision(
 	foundWithoutSubFilter := false
 
 	// Process all placement bindings without subFilter
-	for _, pb := range pbList.Items {
+	for i, pb := range pbList.Items {
 		if pb.SubFilter == policiesv1.Restricted {
 			continue
 		}
 
-		found, err := r.isSingleClusterInDecisions(ctx, &pb, rootPlc.GetName(), clusterName)
+		found, err := r.isSingleClusterInDecisions(ctx, &pbList.Items[i], rootPlc.GetName(), clusterName)
```

Comment:
Is this for saving memory?

Reply:
This is the vulnerability/smell that SonarCloud noticed.

When you take the address of a loop variable like this was doing, you don't get what you'd necessarily expect. Golang kind of re-uses the loop variable in different iterations, so you end up with the same address each time, I think. I believe they want to change it in Go more generally, although I might be confusing some details, and this might not be exactly the same situation: https://go.dev/blog/loopvar-preview

Reply:
oh that... got it!! Good catch